include/ignition/math/Helpers.hh

```diff
@@ -727,6 +729,37 @@ namespace ignition
       }
     }
 
+    /// \brief Convert a std::chrono::steady_clock::time_point to a seconds and
+    /// nanoseconds pair.
+    /// \param[in] _time The time point to convert.
+    /// \return A pair where the first element is the number of seconds and
+    /// the second is the number of nanoseconds.
+    inline std::pair<int64_t, int64_t> timePointToSecNsec(
+        const std::chrono::system_clock::time_point &_time)
+    {
+      auto now_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(_time.time_since_epoch());
+      auto now_s = std::chrono::duration_cast<std::chrono::seconds>(_time.time_since_epoch());
+      int64_t seconds = std::chrono::duration_cast<std::chrono::seconds>(_time.time_since_epoch()).count();
+      int64_t nanoseconds = static_cast<int32_t>(std::chrono::duration_cast<std::chrono::duration<float>>
+        (now_ns - now_s).count()*1e9);
```

Comment:
I feel like there's a more precise way to do this conversion. Does the following work?

~~~
int64_t nanoseconds = std::chrono::duration_cast<std::chrono::nanoseconds>(now_ns - now_s).count();
~~~

Reply:
https://github.com/ignitionrobotics/ign-math/pull/150/commits/b7ccbcdc26c682b31d57612ac8979fa164ba6fc2